flatc generates constructors with memset for non-trivial types

https://github.com/google/flatbuffers/blob/b99332efd732e6faf60bb7ce1ce5902ed65d5ba3/src/idl_gen_cpp.cpp#L2640-L2643

The generated code results in compiler warning/errors on GCC 9.

```
> BUILDSTDERR: In file included from /builddir/build/BUILD/flatbuffers-1.10.0/tests/test.cpp:32:
> BUILDSTDERR: /builddir/build/BUILD/flatbuffers-1.10.0/tests/monster_test_generated.h: In constructor 'MyGame::Example::Vec3::Vec3()':
> BUILDSTDERR: /builddir/build/BUILD/flatbuffers-1.10.0/tests/monster_test_generated.h:315:33: error: 'void* memset(void*, int, size_t)' clearing an object of non-trivial type 'struct MyGame::Example::Vec3'; use assignment or value-initialization instead [-Werror=class-memaccess]
> BUILDSTDERR:   315 |     memset(this, 0, sizeof(Vec3));
> BUILDSTDERR:       |                                 ^
> BUILDSTDERR: /builddir/build/BUILD/flatbuffers-1.10.0/tests/monster_test_generated.h:301:41: note: 'struct MyGame::Example::Vec3' declared here
> BUILDSTDERR:   301 | FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(16) Vec3 FLATBUFFERS_FINAL_CLASS {
> BUILDSTDERR:       |                                         ^~~~
> BUILDSTDERR: cc1plus: all warnings being treated as errors
> BUILDSTDERR: make[2]: *** [CMakeFiles/flattests.dir/build.make:148: CMakeFiles/flattests.dir/tests/test.cpp.o] Error 1
```

--- a/samples/monster_generated.h
+++ b/samples/monster_generated.h
@@ -166,7 +166,7 @@ FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) Vec3 FLATBUFFERS_FINAL_CLASS {
 
  public:
   Vec3() {
-    memset(this, 0, sizeof(Vec3));
+    memset(static_cast<void *>(this), 0, sizeof(Vec3));
   }
   Vec3(float _x, float _y, float _z)
       : x_(flatbuffers::EndianScalar(_x)),

--- a/src/idl_gen_cpp.cpp
+++ b/src/idl_gen_cpp.cpp
@@ -2617,7 +2617,7 @@ class CppGenerator : public BaseGenerator {
 
     // Generate a default constructor.
     code_ += "  {{STRUCT_NAME}}() {";
-    code_ += "    memset(this, 0, sizeof({{STRUCT_NAME}}));";
+    code_ += "    memset(static_cast<void *>(this), 0, sizeof({{STRUCT_NAME}}));";
     code_ += "  }";
 
     // Generate a constructor that takes all fields as arguments.

--- a/tests/monster_test_generated.h
+++ b/tests/monster_test_generated.h
@@ -516,7 +516,7 @@ FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(2) Test FLATBUFFERS_FINAL_CLASS {
 
  public:
   Test() {
-    memset(this, 0, sizeof(Test));
+    memset(static_cast<void *>(this), 0, sizeof(Test));
   }
   Test(int16_t _a, int8_t _b)
       : a_(flatbuffers::EndianScalar(_a)),
@@ -559,7 +559,7 @@ FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(8) Vec3 FLATBUFFERS_FINAL_CLASS {
 
  public:
   Vec3() {
-    memset(this, 0, sizeof(Vec3));
+    memset(static_cast<void *>(this), 0, sizeof(Vec3));
   }
   Vec3(float _x, float _y, float _z, double _test1, Color _test2, const Test &_test3)
       : x_(flatbuffers::EndianScalar(_x)),
@@ -631,7 +631,7 @@ FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) Ability FLATBUFFERS_FINAL_CLASS {
 
  public:
   Ability() {
-    memset(this, 0, sizeof(Ability));
+    memset(static_cast<void *>(this), 0, sizeof(Ability));
   }
   Ability(uint32_t _id, uint32_t _distance)
       : id_(flatbuffers::EndianScalar(_id)),

--- a/tests/namespace_test/namespace_test1_generated.h
+++ b/tests/namespace_test/namespace_test1_generated.h
@@ -57,7 +57,7 @@ FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) StructInNestedNS FLATBUFFERS_FINAL_CLASS 
 
  public:
   StructInNestedNS() {
-    memset(this, 0, sizeof(StructInNestedNS));
+    memset(static_cast<void *>(this), 0, sizeof(StructInNestedNS));
   }
   StructInNestedNS(int32_t _a, int32_t _b)
       : a_(flatbuffers::EndianScalar(_a)),

--- a/tests/union_vector/union_vector_generated.h
+++ b/tests/union_vector/union_vector_generated.h
@@ -189,7 +189,7 @@ FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) Rapunzel FLATBUFFERS_FINAL_CLASS {
 
  public:
   Rapunzel() {
-    memset(this, 0, sizeof(Rapunzel));
+    memset(static_cast<void *>(this), 0, sizeof(Rapunzel));
   }
   Rapunzel(int32_t _hair_length)
       : hair_length_(flatbuffers::EndianScalar(_hair_length)) {
@@ -214,7 +214,7 @@ FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) BookReader FLATBUFFERS_FINAL_CLASS {
 
  public:
   BookReader() {
-    memset(this, 0, sizeof(BookReader));
+    memset(static_cast<void *>(this), 0, sizeof(BookReader));
   }
   BookReader(int32_t _books_read)
       : books_read_(flatbuffers::EndianScalar(_books_read)) {

